versioned_docs/version-v2/components/styling.md

```diff
@@ -233,6 +233,70 @@ By specifying "inner-text" as the value of the `exportparts` attribute, elements
 <outer-component />
 ```
 
+## Style Modes
+
+Component Style Modes enable you to create versatile designs for your components by utilizing different styling configurations. This is achieved by assigning the styleUrls property of a component to a collection of style mode names, each linked to their respective CSS files.
+
+### Example: Styling a Button Component
+
+Consider a basic button component that supports both iOS and Material Design aesthetics:
+
+```tsx title="Using style modes to style a component"
+@Component({
+  tag: 'simple-button',
+  styleUrls: {
+    md: './simple-button.md.css', // styles for Material Design
+    ios: './simple-button.ios.css' // styles for iOS
+  },
+})
+export class SimpleButton {
+  // ...
+}
+```
+
+In the example above, two different modes are declared. One mode is named `md` (for 'Material Design') and refers back to a Material Design-specific stylesheet. Likewise, the other is named `ios` (for iOS) and references a different stylesheet for iOS-like styling. Both stylesheets are relative paths to the file that declares the component. While we have chosen short names in the above example, there's no limitation to the keys used in the `styleUrls` object.
+
+To dictate the style mode (Material Design or iOS) in which the button should be rendered, you must initialize the desired mode before any component rendering occurs. This can be done as follows:
+
+```ts
+import { setMode } from '@stencil/core';
+setMode(() => 'ios'); // Setting iOS as the default mode for all components
+```
+
+The `setMode` function processes all elements, enabling the assignment of modes individually based on specific element attributes. For instance, by assigning the `colorMode` attribute to a component:
+
+```html
+<simple-button colorMode="ios"></simple-button>
+```
+
+You can conditionally set the style mode based on the `colorMode` property:
+
+```ts
+import { setMode } from '@stencil/core';
+
+const defaultMode = 'md'; // Default to Material Design
+setMode((el) => el.getAttribute('colorMode') || defaultMode);
+```
```

Comment:
Double checking something - can you point me to where we check `colorMode` in Stencil?

Reply:
We don't, this is just an example. I learned today that the Framework team uses `mode` as property name for essentially the same. It is not required to use any specific property name here.

Reply:
Let me update to just use `mode` here as `color` can be confusing.

Reply:
Thanks! LGTM